Source/Core/DiscIO/TGCBlob.h

```diff
@@ -0,0 +1,68 @@
+// Copyright 2016 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <memory>
+#include <utility>
+
+#include "Common/CommonTypes.h"
+#include "Common/FileUtil.h"
+#include "DiscIO/Blob.h"
+
+namespace DiscIO
+{
+bool IsTGCBlob(const std::string& path);
+
+struct TGCHeader
+{
+  u32 magic;
+  u32 unknown_1;
+  u32 header_size;
+  u32 unknown_2;
+
+  u32 fst_offset;
+  u32 fst_size;
+  u32 fst_max_size;
+  u32 dol_offset;
+
+  u32 dol_size;
+  u32 unknown_important_1;
+  u32 unknown_3;
+  u32 unknown_4;
+
+  u32 unknown_5;
+  u32 unknown_important_2;
+};
+
+class TGCFileReader : public IBlobReader
+{
+public:
+  static std::unique_ptr<TGCFileReader> Create(const std::string& filename);
+
+  BlobType GetBlobType() const override { return BlobType::TGC; }
+  u64 GetDataSize() const override { return m_size; }
+  u64 GetRawSize() const override { return m_size; }
+  bool Read(u64 offset, u64 nbytes, u8* out_ptr) override;
+
+private:
+  TGCFileReader(File::IOFile&& file);
+
+  bool InternalRead(u64 offset, u64 nbytes, u8* out_ptr);
+
+  static u32 SubtractBE32(u32 minuend_be, u32 subtrahend_le);
```

Comment:
These static functions can just be put in an anonymous namespace in the cpp file, since they're implementation details.